### ISSUE TITLE
importHelpers not respected when exporting defaults

### ISSUE THREAD

**Issue by jordanrastrick** [microsoft/TypeScript#58827]
### 🔎 Search Terms

"importHelpers"

"default exports"

"tslib"

### 🕗 Version & Regression Information

- This is a crash
- This changed between versions ______ and _______
- This changed in commit or PR _______
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about _________
- I was unable to test this on prior versions because _______

### ⏯ Playground Link

Can't create playground as multiple files are required to demonstrate

### 💻 Code

`src/exporter.ts`:
```ts
export default function foo(): boolean {
    return true
}
```
`src/index.ts`:
```ts
export { default as foo } from './exporter'
```

### 🙁 Actual behavior

With `importHelpers` set to `true` and `noEmitHelpers` set to false I expect `src/index.ts` to be compiled to something like:

```js
"use strict";
const tslib_1 = require("tslib");
Object.defineProperty(exports, "__esModule", { value: true });
exports.foo = void 0;
var exporter_1 = require("./exporter");
Object.defineProperty(exports, "foo", { enumerable: true, get: function () { return tslib_1.__importDefault(exporter_1).default; } });
```

### 🙂 Expected behavior

`src/index.ts` is actually compiled to:

```js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.foo = void 0;
var exporter_1 = require("./exporter");
Object.defineProperty(exports, "foo", { enumerable: true, get: function () { return __importDefault(exporter_1).default; } });
```

i.e. it appears for an `export { default as ... }` statement, the `importHelpers` property is being ignored.

**Comment by jakebailey** [comment]
This does show in the playground: [Playground Link](https://www.typescriptlang.org/play/?noEmitHelpers=true&importHelpers=true&module=1#code/KYDwDg9gTgLgBAbzgE2AMwIYFcA28MDOcaEEcAvsVBALZwDkAdAPSiSzBT1A)

But, it's also fixed in 5.5.

**Comment by jakebailey** [comment]
Fix was in #57220, so this is a dupe of #40328.
